### ISSUE TITLE
PSR12.Files.DeclareStatement.SpaceFoundBeforeDirectiveValue

### ISSUE THREAD

**Issue by LLyaudet** [PHPCSStandards/PHP_CodeSniffer#1402]
## Describe the bug

PSR12.Files.DeclareStatement.SpaceFoundBeforeDirectiveValue yields a false positive when using
```php
declare(encoding='ISO-8859-1');
```
Expected no space between equal sign and the directive value in a declare statement

### Code sample
```php
<?php
declare(encoding='ISO-8859-1');
$a = 1;
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs --report=code --standard=PSR12 --sniffs=PSR12.Files.DeclareStatement test.php`
3. See error message displayed
```text
FILE: test.php
-----------------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
-----------------------------------------------------------------------------------------------------------
LINE 1: ERROR Expected no space between the closing parenthesis and the semicolon in a declare statement
-----------------------------------------------------------------------------------------------------------
>> 1:  <?php
   2:  declare(encoding='ISO-8859-1');
   3:  $a·=·1;
-----------------------------------------------------------------------------------------------------------
LINE 2: ERROR Expected no space between equal sign and the directive value in a declare statement
-----------------------------------------------------------------------------------------------------------
   1:  <?php
>> 2:  declare(encoding='ISO-8859-1');
   3:  $a·=·1;
-----------------------------------------------------------------------------------------------------------
LINE 3: ERROR Expected no space between the directive value and closing parenthesis in a declare statement
-----------------------------------------------------------------------------------------------------------
   1:  <?php
   2:  declare(encoding='ISO-8859-1');
>> 3:  $a·=·1;
-----------------------------------------------------------------------------------------------------------
Time: 28ms; Memory: 6MB

```

## Expected behavior

None of these errors.

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System            | Ubuntu 25.10                                                          |
| PHP version                      | 8.4                                                                                         |
| PHP_CodeSniffer version | 4.0.1                                                                                  |
| Standard                          | PSR12                                                         |
| Install type                       | composer global |

## Additional context

Looking at the code here:
<https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/9dfea15a05f25a3e64e21f61ab6bf4b88209a05e/src/Standards/PSR12/Sniffs/Files/DeclareStatementSniff.php#L119-L134>
It seems that a token LNUMBER is expected because tick and strict_types needs a number value.
But encoding doesn't and isn't supported yet.

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.

**Comment by rodrigoprimo** [comment]
I confirmed this issue.

PR #578 is an effort to rewrite this sniff, and it fixes the issue reported here, among other things. However, that PR has not been updated in a while.

@jrfnl and @fredden, fixing this particular issue seems straightforward. Would it be worth preparing a targeted PR, or is it better to wait for #578 to be updated?
